src/map_memory.h

```diff
@@ -97,7 +98,7 @@ struct mm_submap {
  * and each region is saved in its own file.
  */
 struct mm_region {
-    shared_ptr_fast<mm_submap> submaps[MM_REG_SIZE][MM_REG_SIZE];
+    cata::mdarray<shared_ptr_fast<mm_submap>, point, MM_REG_SIZE, MM_REG_SIZE> submaps;
```

Comment:
Seems this is causing build errors in experimental release: https://github.com/CleverRaven/Cataclysm-DDA/runs/7715084204?check_suite_focus=true#step:14:198
```
src/map_memory.cpp: In constructor 'mm_region::mm_region()':
src/map_memory.cpp:52:1: error: function might be candidate for attribute 'noreturn' [-Werror=suggest-attribute=noreturn]
   52 | mm_region::mm_region() : submaps {{ nullptr }} {}
      | ^~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
make: *** [Makefile:952: objwin/tiles/map_memory.o] Error 1
make: *** Waiting for unfinished jobs....
```